### PR TITLE
openshift: wait for container and API to respond

### LINKED ISSUE #26: OpenShift startup fails (on first start)
```
$ sudo systemctl enable openshift
$ sudo systemctl start openshift
Job for openshift.service failed because the control process exited with error code. See "systemctl status openshift.service" and "journalctl -xe" for details.

$ sudo journalctl --unit=openshift
Feb 15 09:21:36 localhost.localdomain systemd[1]: openshift.service: main process     exited, code=exited, status=143/n/a
Feb 15 09:21:36 localhost.localdomain systemd[1]: Failed to start Docker     Application Container for OpenShift.
Feb 15 09:21:36 localhost.localdomain systemd[1]: Unit openshift.service entered     failed state.
Feb 15 09:21:36 localhost.localdomain systemd[1]: openshift.service failed.
Feb 15 09:22:06 localhost.localdomain systemd[1]: openshift.service holdoff time     over, scheduling restart.
Feb 15 09:22:06 localhost.localdomain systemd[1]: Starting Docker Application     Container for OpenShift...
Feb 15 09:22:06 localhost.localdomain docker[13560]: openshift
Feb 15 09:22:07 localhost.localdomain docker[13565]: openshift
Feb 15 09:22:07 localhost.localdomain docker[13571]: Error response from daemon:     Conflict: Tag latest is already set to image     62208f15133798af253a651386ae2753da9130fcc68f1863cbd735c547356173, if you want to     replace it, please use -f opt
Feb 15 09:22:08 localhost.localdomain sh[13667]: [INFO] Starting OpenShift server    
```

### PR DESCRIPTION
**Pull request by vrutkovs**
Fixes #26

### CODE REVIEW

File: services/openshift/openshift.service [hunk 1]
```diff
@@ -14,7 +14,6 @@ ExecStartPre=-/usr/bin/docker rm openshift
 ExecStartPre=-/usr/bin/docker tag ${DOCKER_REGISTRY}/${IMAGE} openshift
 ExecStartPre=-/usr/bin/sh /opt/adb/openshift/docker_config $OPTIONS
 ExecStart=/usr/bin/sh /opt/adb/openshift/openshift $OPTIONS
-ExecStartPost=/bin/sleep 30
```
Comment (hferentschik) [inline services/openshift/openshift.service:17]: +1 for removing this. This timeout might or might not work based on your machine. We need to actively check until openshift is up

File: services/openshift/scripts/openshift_provision [hunk 2]
```diff
@@ -9,6 +9,19 @@ export ORIGIN_DIR="/var/lib/origin"
 export OPENSHIFT_DIR=${ORIGIN_DIR}/openshift.local.config/master
 export KUBECONFIG=${OPENSHIFT_DIR}/admin.kubeconfig
 
+# Wait for container to show up
+ATTEMPT=0
```
Comment (hferentschik) [inline services/openshift/scripts/openshift_provision]: Should this not really be in the openshift script? Once provision is called I would expect that OpenShift is up and running

File: services/openshift/scripts/openshift_provision [hunk 3]
```diff
@@ -9,6 +9,19 @@ export ORIGIN_DIR="/var/lib/origin"
...
+until docker inspect openshift > /dev/null 2>&1 || [ $ATTEMPT -eq 10 ]; do
+  sleep 1
+  ((ATTEMPT++))
+done
+
+ATTEMPT=0
+until curl -ksSf https://127.0.0.1:8443/api > /dev/null 2>&1 || [ $ATTEMPT -eq 30 ]; do
```
Comment (hferentschik) [inline services/openshift/scripts/openshift_provision]: I still think we should use the internal OpenShift cluster IP here

File: services/openshift/scripts/openshift_provision (hunk 2 above)
Comment (vrutkovs) [inline services/openshift/scripts/openshift_provision]: Yeah, I can move it to `openshift`, it doesn't matter much

Comment (hferentschik) [inline services/openshift/scripts/openshift_provision]: Sure, but it seems to be better of there I think.

File: services/openshift/scripts/openshift_provision (hunk 3 above)
Comment (LalatenduMohanty) [inline services/openshift/scripts/openshift_provision]: @hferentschik +1

**Comment by vrutkovs** [comment]
Right, created `wait_for_openshift_api` in `openshift`, leaving a final check in `openshift_provision`

**Comment by navidshaikh** [comment]
LGTM. Wondering if we should echo a message after certain `ATTEMPT` stating "waiting for server to configure" (or something similar).

**Comment by hferentschik** [comment]
+1

**Comment by LalatenduMohanty** [comment]
LGTM

**Pull request merged**